File: odemapster-core/src/main/java/es/upm/fi/dia/oeg/obdi/core/sql/SQLJoinTable.java
package es.upm.fi.dia.oeg.obdi.core.sql;

import java.util.Collection;
import java.util.Vector;

import Zql.ZExp;
import Zql.ZExpression;
import Zql.ZFromItem;
import es.upm.fi.dia.oeg.morph.base.Constants;
import es.upm.fi.dia.oeg.morph.base.MorphSQLUtility;

@SuppressWarnings("serial")
public class SQLJoinTable extends ZFromItem {
//	private static Logger logger = Logger.getLogger(SQLJoinTable.class);
	
	private SQLLogicalTable joinSource;
	private String joinType;
	private ZExpression onExpression;

	public SQLJoinTable(SQLLogicalTable joinSource,String joinType, ZExpression onExp) {
		super();
		this.joinType = joinType;
		this.joinSource = joinSource;
		this.onExpression = onExp;
	}

	public SQLJoinTable(SQLLogicalTable joinSource) {
		this(joinSource, null, null);
	}
	
	public void setOnExpression(ZExpression onExp) {
		this.onExpression = onExp;
	}

	public void addOnExpression(ZExpression onExp2) {
		Collection<ZExp> expressionsList = new Vector<ZExp>();
		if(this.onExpression != null) {expressionsList.add(this.onExpression);}
		if(onExp2 != null) {expressionsList.add(onExp2);}
		this.onExpression = MorphSQLUtility.combineExpresions(expressionsList
				, Constants.SQL_LOGICAL_OPERATOR_AND());
	}
	
	public void setJoinType(String joinType) {
		this.joinType = joinType;
	}

	public void setJoinSource(SQLFromItem fromItem) {
		this.joinSource = fromItem;
	}

	@Override
	public String toString() {
		StringBuffer result = new StringBuffer();
		if(this.joinType != null) {
			result.append(this.joinType + " JOIN ");	
		}
		result.append(this.joinSource);
		if(this.onExpression != null) {
			result.append(" ON " + this.onExpression);	
		}
		
		return result.toString();
	}

	public SQLLogicalTable getJoinSource() {
		return joinSource;
	}

	public String getJoinType() {
		return joinType;
	}

	public ZExpression getOnExpression() {
		return onExpression;
	}
}

File: morph-base/src/main/scala/es/upm/fi/dia/oeg/morph/base/sql/SQLJoinTable.scala
package es.upm.fi.dia.oeg.morph.base.sql

import scala.collection.JavaConversions._
import Zql.ZFromItem
import Zql.ZExpression
import org.apache.log4j.Logger
import Zql.ZExp
import es.upm.fi.dia.oeg.morph.base.Constants

class SQLJoinTable(val joinSource:SQLLogicalTable ,val joinType:String 
    , var onExpression:ZExpression ) extends ZFromItem {
	val logger = Logger.getLogger(this.getClass().getName());
	
	def this(joinSource:SQLLogicalTable ) = {
		this(joinSource, null, null);
	}
	
	def addOnExpression(onExp2:ZExpression ) = {
		val expressionsList = Set(this.onExpression, onExp2);
		
		this.onExpression = MorphSQLUtility.combineExpresions(
		    expressionsList, Constants.SQL_LOGICAL_OPERATOR_AND);
	}
	
	override def toString() : String  = {
		var result = "";
		if(this.joinType != null) {
			result += this.joinType + " JOIN ";	
		}
		result += this.joinSource;
		if(this.onExpression != null) {
			result += " ON " + this.onExpression;	
		}
		
		result;
	}


}